source/llassetgen-rendering/main.cpp

```diff
@@ -0,0 +1,383 @@
+
+    auto colorValidator = new QIntValidator(0, 255);
+
+    // FONT COLOR
+    auto fontColorGroupBox = new QGroupBox("Font Color");
+    fontColorGroupBox->setMaximumHeight(150);
+    auto fontColorLayout = new QFormLayout();
+
+    fontColorGroupBox->setLayout(fontColorLayout);
+
+    // font Color RED
+    auto *fontR = new QLineEdit();
+    auto *labelFontR = new QLabel("R:");
+    labelFontR->setMaximumWidth(90);
+    fontR->setValidator(colorValidator);
+    fontR->setPlaceholderText("0");
+    fontR->setMaximumWidth(45);
+
+    QObject::connect(fontR, SIGNAL(textEdited(QString)), glwindow, SLOT(fontColorRChanged(QString)));
+    fontColorLayout->addRow(labelFontR, fontR);
+
+    // font Color GREEN
+    auto *fontG = new QLineEdit();
+    auto *labelFontG = new QLabel("G:");
+    labelFontG->setMaximumWidth(90);
+    fontG->setValidator(colorValidator);
+    fontG->setPlaceholderText("0");
+    fontG->setMaximumWidth(45);
+
+    QObject::connect(fontG, SIGNAL(textEdited(QString)), glwindow, SLOT(fontColorGChanged(QString)));
+    fontColorLayout->addRow(labelFontG, fontG);
+
+    // font Color BLUE
+    auto *fontB = new QLineEdit();
+    auto *labelFontB = new QLabel("B:");
+    labelFontB->setMaximumWidth(90);
+    fontB->setValidator(colorValidator);
+    fontB->setPlaceholderText("0");
+    fontB->setMaximumWidth(45);
+
+    QObject::connect(fontB, SIGNAL(textEdited(QString)), glwindow, SLOT(fontColorBChanged(QString)));
+    fontColorLayout->addRow(labelFontB, fontB);
+
+    // BACKGROUND COLOR
+
+    auto backgroundColorGroupBox = new QGroupBox("Background Color");
+    backgroundColorGroupBox->setMaximumHeight(150);
+    auto backgroundColorLayout = new QFormLayout();
+    backgroundColorGroupBox->setLayout(backgroundColorLayout);
+
+    // Background Color RED
+    auto *backgroundR = new QLineEdit();
+    auto *labelR = new QLabel("R");
+    labelR->setMaximumWidth(90);
+    backgroundR->setValidator(colorValidator);
+    backgroundR->setPlaceholderText("255");
+    backgroundR->setMaximumWidth(45);
+
+    QObject::connect(backgroundR, SIGNAL(textEdited(QString)), glwindow, SLOT(backgroundColorRChanged(QString)));
+
+    backgroundColorLayout->addRow(labelR, backgroundR);
+
+    // Background Color GREEN
+    auto *backgroundG = new QLineEdit();
+    auto *labelG = new QLabel("G");
+    labelG->setMaximumWidth(90);
+    backgroundG->setValidator(colorValidator);
+    backgroundG->setPlaceholderText("255");
+    backgroundG->setMaximumWidth(45);
+
+    QObject::connect(backgroundG, SIGNAL(textEdited(QString)), glwindow, SLOT(backgroundColorGChanged(QString)));
+
+    backgroundColorLayout->addRow(labelG, backgroundG);
+
+    // Background Color BLUE
+    auto *backgroundB = new QLineEdit();
+    auto *labelB = new QLabel("B");
+    labelB->setMaximumWidth(90);
+    backgroundB->setValidator(colorValidator);
+    backgroundB->setPlaceholderText("255");
+    backgroundB->setMaximumWidth(45);
```

Comment:
Could probably made shorter by making a factory methods for color fields

Reply:
A factory (as in GoF) needs subclasses, I think that's too much here. But I created a method that replaces 3 lines and makes maintenance easier.